core/src/main/java/smile/gap/BitString.java

```diff
@@ -192,7 +192,27 @@ public BitString(int[] bits, FitnessMeasure<BitString> measure, Crossover crosso
     public int compareTo(Chromosome o) {
         return (int) Math.signum(fitness - o.fitness());
     }
-    
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj instanceof Chromosome) {
+            return this.fitness == ((Chromosome)obj).fitness();
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
this doesn't make sense. all other fields are ignored. and I don't expect BitString to be used in a hash map.